lib/topological_inventory/openshift/operations/core/authentication_retriever.rb

```diff
@@ -1,16 +1,22 @@
+require "topological_inventory/openshift/operations/core/retriever"
+
 module TopologicalInventory
   module Openshift
     module Operations
       module Core
-        class AuthenticationRetriever
-          def initialize(endpoint_id)
-            @endpoint_id = endpoint_id
-          end
-
+        class AuthenticationRetriever < Retriever
           def process
-            # Without using Rails example:
-            # https://github.com/ManageIQ/topological_inventory-orchestrator/pull/1/files#diff-4743ee12ee7e621468f2e6590de994efR97
-            # Authentication.where(:resource_type => "Endpoint", :resource_id => @endpoint_id).first
+            authentication_id = @api_instance.list_endpoint_authentications(@id.to_s).data[0].id
```

Comment:
Can you get the authentication_id in the caller so that `@id` is the authentication_id for consistency with the rest of the Retrievers?

Reply:
And if there are no authentications on the endpoint this should probably do `.data.first&.id` and return if authentication_id is inl